keps/sig-node/1769-memory-manager/README.md

```diff
@@ -0,0 +1,782 @@
+
+The new `--reserved-memory` was added to allow for this total reserved memory to be split (by a node administrator) across many NUMA nodes. 
+
+Syntax:
+* `--reserved-memory=[{numa-node=int,type=string,limit=string}][,][...]`
+* `numa-node` - an integer index of NUMA node, e.g. `0`
+* `type` can be either set to:
+  * `memory` to reserve conventional memory
+  * `hugepages-2Mi` or `hugepages-1Gi` to reserve hugepages 
+* `limit` - the amount of reserved memory, e.g. `1Gi`
+
+Example A:
+
+`--reserved-memory={numa-node=0,type=memory,limit=1Gi},{numa-node=1,type=memory,limit=2Gi}`
+
+The following formula must hold for each memory type: 
+
+`sum(SystemReserved(i)) = kube-reserved + system-reserved + eviction-threshold`, 
+
+where `i` is an index of a NUMA node, while the values above can be specified with Node Allocatable Feature flags (more details [here](https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/)), for example:
+* `--kube-reserved=cpu=500m,memory=50Mi`
+* `--system-reserved=cpu=123m,memory=333Mi`
+* `--eviction-hard=memory.available<500Mi`
+
+If the above formula does not hold, the Memory Manager returns an error.
+
+In other words, "Example A" indicates that for conventional memory (`type=memory`) we reserve in total: 
+
+`sum(SystemReserved(i)) = SystemReserved(0) + SystemReserved(1) = 1Gi + 2Gi = 3Gi`
+
+NOTICE:
+- The administrator guide is out of scope of this proposal. It is assumed that node administrators have ample professional background, and therefore, they are able to reserve suitable amount of memory across all NUMA nodes so that system and kubelet can operate in a stable way on NUMA-based servers. 
+
+<!-- ### The configuration of Memory Manager
+
+Following flags are used to configure the Memory Manager:
+- Feature Gate: `--feature-gate=MemoryManager=true|false`
+- Kubelet Flag: `--reserved-memory=[{numa-node=int, type=string, limit=string}][,][...]`
+
+#### Feature Gate Flag
+
+#### Reserved Memory Flag
+-->
+
+<!-- 
+As described in Memory Map section, the size of Memory Reserved and Allocatable Zone are determined by the configuration of Memory Manager. Administrators are expected to have high confidence in the configuration of `--reserved-memory` flag.
+As mentioned above, This feature will provide basic validation of configuration for administrators.
+
+The Node Allocatable memory of Node Allocatable Feature is calculated by following formula:
+- [Node-Allocatable] = [Node-Capacity] - [Kube-Reserved] - [System-Reserved] - [Eviction-Threshold]
+
+The Node Allocatable is exposed to API server as part of `v1.Node` object and referred by scheduler to select appropriate worker node to bind a Pod.
+
+NOTE:
+- With respect to hugepages, in v1.17, Node Allocateble feature did not support the reservation of hugepages. Once Node Allocatable feature supports hugepages reservation (#83541), Reserved Memory for hugepages will operate in the same manner as for the conventional memory.
```

Comment:
@bg-chun Could you reply?

Reply:
@derekwaynecarr 
There is the OVS-DPDK which is a virtualized switch for high-performance switching.
NFV guys are still running that in host space, intel guys contributed [userspace-cni](https://github.com/intel/userspace-cni-network-plugin)(which allows us to run OVS-DPDK in a Pod) though.

If OVS-DPDK is deployed in a node, it will take some hugepages, and there should be a way to tell kubelet how many hugepages were(and will be) taken.